lib/puppet/provider/oneview_san_manager/c7000.rb

```diff
@@ -14,32 +14,27 @@
 # limitations under the License.
 ################################################################################
 
-require_relative '../login'
-require_relative '../common'
-require 'oneview-sdk'
+require_relative '../oneview_resource'
+
+Puppet::Type::Oneview_san_manager.provide :c7000 do
+  desc 'Provider for OneView SAN Manager using the C7000 variant of the OneView API'
+
+  confine true: login[:hardware_variant] == 'C7000'
 
-Puppet::Type.type(:oneview_san_manager).provide(:oneview_san_manager) do
   mk_resource_methods
 
   def initialize(*args)
     super(*args)
     @client = OneviewSDK::Client.new(login)
-    @resourcetype = OneviewSDK::SANManager
+    api_version = login[:api_version] || 200
+    @resourcetype ||= if api_version == 200
```

Comment:
should use ``def resource_name 'SANManager' end`` and ``def self.resource_name 'SANManager' end`` instead